### PR TITLE
Support generichide rules

### PR DESCRIPTION
**Pull request by antonok-edm**
Adds support for `$generichide` exception rules (as well as the corresponding uBO-style alias, `$ghide`). These are network rules that prevent generic cosmetic element hiding from being applied to matching URLs.

This partially addresses https://github.com/brave/adblock-rust/issues/1.

### CODE REVIEW

File: src/blocker.rs [hunk 1]
```diff
@@ -1346,14 +1389,13 @@ mod legacy_rule_parsing_tests {
     // easyList = { 24478, 31144, 0, 5589 };
     // not handling (and not including) filters with the following options: 
     // - $popup
-    // - $generichide
     // - $subdocument
```
Comment (AndriusA) [inline src/blocker.rs:1392]: not about this PR, but I think we actually do handle `$subdocument`

Comment (antonok-edm) [reply src/blocker.rs:1392]: looks like `$document` right below is handled too. I'll go ahead and remove both.

Comment (AndriusA) [reply src/blocker.rs]: `$document` only somewhat - there is a spec for handling main document url blocking, but not implemented

Comment (antonok-edm) [reply src/blocker.rs]: ah oops, I'll add that back then :sweat_smile:

**Pull request merged**